### ISSUE TITLE
MSTest.TestAdapter should target newer TargetFrameworks

### ISSUE THREAD

**Issue by ericstj** [microsoft/testfx#1073]
## Description
Today MSTest.TestAdapter targets old and out of support target frameworks, resulting in a lot of unnecessary dependencies being downloaded.

## Steps to reproduce
Reference MSTest.TestAdapter in a `net6.0` project.

## Expected behavior
No additional packages downloaded since all the API needed by MSTest.TestAdapter is present in the framework.

## Actual behavior
Many packages downloaded (100s) due to using the older package dependency groups that bring down the old `NETStandard.Library` package reference.

This can also be observed in the `project.assets.json`

## Environment
Repro project:
```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net6.0</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="MSTest.TestAdapter" Version="2.2.9" />
  </ItemGroup>

</Project>
```

Repro steps
1. dotnet restore
2. examine `project.assets.json`

**Comment by Haplois** [comment]
Thank you for reporting this, we are aware of this issue. Unfortunately we cannot break compatibility with older TFMs, so we are adding newer frameworks to our packages slowly. I'll keep this assigned to myself so I can monitor its progress.

**Comment by SymbioticKilla** [comment]
Perhaps it is better to solve it with V3 and v2 leave for hotfixes?

**Comment by Haplois** [comment]
Yes, that's the plan.

**Comment by Evangelink** [comment]
This will be available in 2.3.0
